core/services/bag_of_holding/main.py

```diff
@@ -52,21 +54,30 @@ def write_db(data: Dict[str, Any]) -> None:
         json.dump(data, f)
 
 
-@app.post("/set/{path}")
+@app.post("/overwrite")
+async def overwrite_data(payload: dict[str, Any] = Body(...)) -> JSONResponse:
+    logger.debug(f"Overwrite: {json.dumps(payload)}")
+    write_db(payload)
+    return JSONResponse(content={"status": "success"})
+
+
+@app.post("/set/{path:path}")
 @version(1, 0)
-async def write_data(path: str, payload: dict[str, Any] = Body(...)) -> JSONResponse:
+async def write_data(path: str = FastPath(..., regex=r"^.*$"), payload: dict[str, Any] = Body(...)) -> JSONResponse:
+    logger.debug(f"Write path: {path}, {json.dumps(payload)}")
     current_data = read_db()
-    dpath.new(current_data, path, payload, separator=".")
+    dpath.new(current_data, path, payload)
     write_db(current_data)
     return JSONResponse(content={"status": "success"})
 
 
-@app.get("/get/{path}")
+@app.get("/get/{path:path}")
 @version(1, 0)
 async def read_data(path: str) -> JSONResponse:
+    logger.debug(f"Get path: {path}")
     current_data = read_db()
     try:
-        result = dpath.get(current_data, path, separator=".")
+        result = dpath.util.search(current_data, path)
```

Comment:
I'm not sure I like this, what is the purpose of using search? I think if we have something like
```
{
  "potato: {
    "a": 1,
    "b": 2
    }
}

and we ask for "potato/a" we should just get "1", instead I'm getting:

```
{
  "potato: {
    "a": 1
    }
}

```

Reply:
Check now

![image](https://user-images.githubusercontent.com/1215497/230996025-04c78be9-d38b-42d3-9f7e-ed310d20a128.png)